### ISSUE TITLE
Why DataCollatorForLanguageModeling do not make attention_mask feature?

### ISSUE THREAD

**Issue by makailove123** [huggingface/transformers#4702]
# ❓ Questions & Help

## Details

As we know, according to padding on batch of input_ids, we should mask the padding ids as invalidate to avoid attention on them. But I found that collate function of DataCollatorForLanguageModeling didn't implement attention_mask feature.
Is this expected? Or do you think attention_mask is not necessary?

**A link to original question on Stack Overflow**:

**Comment by liebkne** [comment]
@makailove123 Have you figured out why?
